datarepo-clienttests/build.gradle

```diff
@@ -96,6 +96,55 @@ task(runTest, dependsOn: 'classes', type: JavaExec) {
     main = "runner.TestRunner"
     classpath = sourceSets.main.runtimeClasspath
 }
+
+def getNamespace(){
+    def pathToServer = "src/main/resources/servers/" + System.env.TEST_RUNNER_SERVER_SPECIFICATION_FILE
+    println pathToServer
+    def serverFile = file(pathToServer)
+    def serverDefinition = new groovy.json.JsonSlurper().parseText(serverFile.text)
+    return serverDefinition.namespace
+}
+
+def runSecretScript(script){
+    def outputStream = new StringBuffer();
+    def proc = Runtime.getRuntime().exec(script)
+    proc.waitForProcessOutput(outputStream, System.err)
+    println(outputStream.toString());
+    return proc.exitValue()
+}
+
+task lockEnv {
+    doFirst {
+        if (System.env.TEST_RUNNER_SERVER_SPECIFICATION_FILE) {
+            def namespace = getNamespace()
+            def secretExitValue = runSecretScript("./tools/namespaceLock.sh $namespace")
+            if (secretExitValue > 0){
+                throw new GradleException("Lock could not be aquired.\n" +
+                    "REASON: Namespace '$namespace' already in use.\n" +
```

Comment:
I kinda wonder if we could find some way to name locks so people could name the locks after themselves to make it easier to see who has the lock

Reply:
Yeah - right now we're relying on the secret name to be the same for everyone. But, I imagine it would be possible. If we run into this enough, I could see this being a follow on.